eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -917,26 +915,33 @@ object InteractiveTxSigningSession {
         }
       case None => None
     }
-    val txWithSigs = FullySignedSharedTransaction(partiallySignedTx.tx, partiallySignedTx.localSigs, remoteSigs, sharedSigs_opt)
-    if (remoteSigs.txId != txWithSigs.signedTx.txid) {
-      log.info("invalid tx_signatures: txId mismatch (expected={}, got={})", txWithSigs.signedTx.txid, remoteSigs.txId)
+    if (remoteSigs.txId != partiallySignedTx.txId) {
+      log.info("invalid tx_signatures: txId mismatch (expected={}, got={})", partiallySignedTx.txId, remoteSigs.txId)
       return Left(InvalidFundingSignature(fundingParams.channelId, Some(partiallySignedTx.txId)))
     }
-    // We allow a 5% error margin since witness size prediction could be inaccurate.
-    if (fundingParams.localContribution != 0.sat && txWithSigs.feerate < fundingParams.targetFeerate * 0.95) {
-      return Left(InvalidFundingFeerate(fundingParams.channelId, fundingParams.targetFeerate, txWithSigs.feerate))
-    }
-    val previousOutputs = {
-      val sharedOutput = fundingParams.sharedInput_opt.map(sharedInput => sharedInput.info.outPoint -> sharedInput.info.txOut).toMap
-      val localOutputs = txWithSigs.tx.localInputs.map(i => i.outPoint -> i.previousTx.txOut(i.previousTxOutput.toInt)).toMap
-      val remoteOutputs = txWithSigs.tx.remoteInputs.map(i => i.outPoint -> i.txOut).toMap
-      sharedOutput ++ localOutputs ++ remoteOutputs
-    }
-    Try(Transaction.correctlySpends(txWithSigs.signedTx, previousOutputs, ScriptFlags.STANDARD_SCRIPT_VERIFY_FLAGS)) match {
-      case Failure(f) =>
-        log.info("invalid tx_signatures: {}", f.getMessage)
+    val txWithSigs = FullySignedSharedTransaction(partiallySignedTx.tx, partiallySignedTx.localSigs, remoteSigs, sharedSigs_opt)
+    txWithSigs.signedTx_opt(fundingParams) match {
+      case Some(signedTx) =>
+        val feerate = Transactions.fee2rate(txWithSigs.tx.fees, signedTx.weight())
+        // We allow a 5% error margin since witness size prediction could be inaccurate.
+        if (fundingParams.localContribution != 0.sat && feerate < fundingParams.targetFeerate * 0.95) {
+          return Left(InvalidFundingFeerate(fundingParams.channelId, fundingParams.targetFeerate, feerate))
+        }
+        val previousOutputs = {
+          val sharedOutput = fundingParams.sharedInput_opt.map(sharedInput => sharedInput.info.outPoint -> sharedInput.info.txOut).toMap
+          val localOutputs = txWithSigs.tx.localInputs.map(i => i.outPoint -> i.previousTx.txOut(i.previousTxOutput.toInt)).toMap
+          val remoteOutputs = txWithSigs.tx.remoteInputs.map(i => i.outPoint -> i.txOut).toMap
+          sharedOutput ++ localOutputs ++ remoteOutputs
+        }
```

Comment:
Maybe move this to `SharedTransaction` right away to minimize diff in feature branch?